test/jdk/javax/swing/plaf/nimbus/TestNimbusTabbedPaneIconPosition.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 6875229
+ * @key headful
+ * @summary  Verifies icon is drawn after text in NimbusL&F
+ * @run main TestNimbusTabbedPaneIconPosition
+ */
+
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Graphics;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import javax.imageio.ImageIO;
+import javax.swing.Icon;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JTabbedPane;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+public class TestNimbusTabbedPaneIconPosition {
+    static JTabbedPane tabbedPane;
+    static JFrame frame;
+    static volatile Point pt;
+    static volatile Rectangle bounds;
+    private static void addTab(JTabbedPane aTabbedPane, int anIndex) {
+        aTabbedPane.addTab("\u2588", new Icon() {
```

Comment:
First time i saw usage of Full block character for title and it took some time for me to understand this, unique way to find difference and read color.

Reply:
> Is there any change in rendering based on what we draw first?

No, since we rely on iconRect and textRect which has its own x and y set so no matter is what called first, it will draw at its own set position